splinter/driver/webdriver/firefox.py

```diff
@@ -29,11 +29,11 @@ def __init__(
         **kwargs
     ):
 
-        firefox_profile = FirefoxProfile(profile)
-        firefox_profile.set_preference("extensions.logging.enabled", False)
-        firefox_profile.set_preference("network.dns.disableIPv6", False)
-
         options = options or Options()
+        if profile:
+            options.set_preference("profile", profile)
```

Comment:
In https://github.com/cobrateam/splinter/issues/1044#issuecomment-1154244607 I've used `options.profile` (after inspecting Selenium's code).